server/mock/app_test.go

```diff
@@ -38,7 +38,12 @@ func TestInitApp(t *testing.T) {
 	req := abci.RequestInitChain{
 		AppStateBytes: appState,
 	}
-	app.InitChain(context.TODO(), &req)
+	res, err := app.InitChain(context.TODO(), &req)
```

Comment:
in this test since we are querying state that is being written to disk then we need to call finalize to push it to disk for commit to flush it